supernode/store/local_storage.go

```diff
@@ -174,7 +174,12 @@ func (ls *localStorage) Put(ctx context.Context, raw *Raw, data io.Reader) error
 	lock(path, raw.Offset, false)
 	defer unLock(path, raw.Offset, false)
 
-	f, err := fileutils.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0644)
+	var f *os.File
+	if raw.Trunc {
+		f, err = fileutils.OpenFile(path, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, 0644)
+	} else {
+		f, err = fileutils.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0644)
```

Comment:
Under what conditions the raw.Trunc bool == false? If not, we can make truncate as the default behavior instead of using a bool

Reply:
In fact, it's a function that unrelated to the business code. And the value of `Truc` should be false when we want to append the file instead of truncate it firstly.

Reply:
Do you mean that raw.Trunc is a function that returns a bool value?

Reply:
It is necessary to make raw.Trunc=false during putting piece data into file. Can we set this value by `raw.Offset` and `raw.Length` ?

Reply:
> Can we set this value by raw.Offset and raw.Length ?

What's that mean? Could you please provide an example?

Reply:
> > Can we set this value by raw.Offset and raw.Length ?
> 
> What's that mean? Could you please provide an example?

I mean that use trunc when `raw.Length`==-1 and `raw.offset`==0 in an implicit form.

Reply:
> In fact, it's a function that unrelated to the business code. And the value of `Truc` should be false when we want to append the file instead of truncate it firstly.

Is it a good idea to also update the unit test code (local_storage_test.go) to reflect this change?